### DISCUSSION TITLE
Why the "validate" nomenclature?

### DISCUSSION THREAD

**Discussion by lijok** [pydantic/pydantic#10746] [General]
Reading through the docs I'm seeing functions like;
```py
def validate_from_int(value: int) -> ThirdPartyType:
    result = ThirdPartyType()
    result.x = value
    return result
```

Which really just looks like a deserializer to me.
I would expect a to not do any conversions, and just raise an exception if the data is invalid.
What's the thinking behind the "validate" / "validators" nomenclature in pydantic?

**Comment by Viicos** [comment]
See the first note in https://docs.pydantic.dev/latest/concepts/models/
